File: data/CodeLlama/readable_format_HPEC/fortran/file236.f90
subroutine print_array(n, x1, x2)
implicit none

DATA_TYPE, dimension(n) :: x1
DATA_TYPE, dimension(n) :: x2
integer :: n
integer :: i
do i = 1, n
write(0, DATA_PRINTF_MODIFIER) x1(i)
write(0, DATA_PRINTF_MODIFIER) x2(i)
if (mod((i - 1), 20) == 0) then
write(0, *)
end if
end do
write(0, *)
end subroutine

File: data/gpt-4o/readable_format_HPEC/cpp/file236.cpp
#include <iostream>
#define DATA_TYPE double
#define DATA_PRINTF_MODIFIER "%f\n"

void print_array(int n, DATA_TYPE* x1, DATA_TYPE* x2)
{
    for (int i = 0; i < n; i++)
    {
        printf(DATA_PRINTF_MODIFIER, x1[i]);
        printf(DATA_PRINTF_MODIFIER, x2[i]);
        if (i % 20 == 0)
        {
            printf("\n");
        }
    }
    printf("\n");
}
